File: 1/2020.09.22/589.N-ary Tree Preorder Traversal/python/solution.py
class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        result = []
        self.traversal(root, result)
        return result

    def traversal(self, root, result):
        if root:
            result.append(root.val)
            if root.children:
                for child in root.children:
                    self.traversal(child, result)

File: 1/2020.09.22/589.N-ary Tree Preorder Traversal/java/solution.java
class Solution {
    public List<Integer> preorder(Node root) {
        List<Integer> res = new ArrayList<>();
        traversal(root, res);
        return res;
    }

    private void traversal(Node root, List<Integer> res) {
        if (root != null) {
            res.add(root.val);
            if (root.children != null) {
                for (Node child : root.children) {
                    traversal(child, res);
                }
            }
        }
    }
}
